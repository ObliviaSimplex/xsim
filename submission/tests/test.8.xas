# =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
# Test 8: Testing Some Known Difficulties with
#         Data Movement between c->memory[c->reg(N)]
#        and c->regs[N].
# =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

.glob main

main:
        std
        push    r14
        mov     r15, r14
        loadi   0xffff, r0
        loadi   storage, r1
        stor    r0, r1
        loadi   8,  r3
        add     r3, r1
        stor    r1, r1
        load    r1, r1
        inc     r1
        load    r1, r1
        inc     r1
        load    r1, r1
        inc     r1
        load    r1, r1
        stor    r0, r1
        load    r0, r0
        mov     r14, r15
        pop     r14
        ret



storage:
.words 0x30