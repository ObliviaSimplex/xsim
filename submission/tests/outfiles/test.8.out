PC: 040c, State: 0002: Registers:
 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0408
PC: 040e, State: 0002: Registers:
 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406
PC: 0410, State: 0002: Registers:
 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406 0406
PC: 0414, State: 0002: Registers:
 ffff 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406 0406
PC: 0418, State: 0002: Registers:
 ffff 043a 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406 0406
PC: 041a, State: 0002: Registers:
 ffff 043a 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406 0406
PC: 041e, State: 0002: Registers:
 ffff 043a 0000 0008 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406 0406
PC: 0420, State: 0002: Registers:
 ffff 0442 0000 0008 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406 0406
PC: 0422, State: 0002: Registers:
 ffff 0442 0000 0008 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406 0406
PC: 0424, State: 0002: Registers:
 ffff 0442 0000 0008 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406 0406
PC: 0426, State: 0002: Registers:
 ffff 0443 0000 0008 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406 0406
PC: 0428, State: 0002: Registers:
 ffff 4200 0000 0008 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406 0406
PC: 042a, State: 0002: Registers:
 ffff 4201 0000 0008 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406 0406
PC: 042c, State: 0002: Registers:
 ffff 0000 0000 0008 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406 0406
PC: 042e, State: 0002: Registers:
 ffff 0001 0000 0008 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406 0406
PC: 0430, State: 0002: Registers:
 ffff f004 0000 0008 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406 0406
PC: 0432, State: 0002: Registers:
 ffff f004 0000 0008 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406 0406
PC: 0434, State: 0002: Registers:
 00e1 f004 0000 0008 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406 0406
PC: 0436, State: 0002: Registers:
 00e1 f004 0000 0008 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406 0406
PC: 0438, State: 0002: Registers:
 00e1 f004 0000 0008 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0408
PC: 0008, State: 0002: Registers:
 00e1 f004 0000 0008 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 040a
CPU has halted.
